cli/command/context.go

```diff
@@ -0,0 +1,44 @@
+package command
+
+import (
+	"github.com/docker/docker/pkg/contextstore"
+)
+
+// ContextMetadata is a typed representation of what we put in Context metadata
+type ContextMetadata struct {
+	Description             string
+	StackOrchestrator       Orchestrator
+	HelperEnabledDockerHost string
+}
+
+// SetContextMetadata set the metadata inside a stored context
+func SetContextMetadata(ctx *contextstore.ContextMetadata, metadata ContextMetadata) {
+	ctx.Metadata = map[string]interface{}{
+		"description":              metadata.Description,
+		"defaultStackOrchestrator": string(metadata.StackOrchestrator),
+		"helperEnabledDockerHost":  metadata.HelperEnabledDockerHost,
+	}
+}
+
+// GetContextMetadata extract metadata from stored context metadata
+func GetContextMetadata(ctx contextstore.ContextMetadata) (ContextMetadata, error) {
+	if ctx.Metadata == nil {
+		return ContextMetadata{}, nil
+	}
+	result := ContextMetadata{}
+	var err error
+	if val, ok := ctx.Metadata["description"]; ok {
+		result.Description, _ = val.(string)
```

Comment:
Didn't I see a string-metadata accessor function in the moby/moby PR?

Reply:
Now that everything lives in the CLI, I will change that to avoid duplication